### ISSUE TITLE
Exporting with only Styles applied does not seem to work.

### ISSUE THREAD

**Issue by jonathanbossenger** [WordPress/create-block-theme#82]
Apologies in advance if this is expected, or if I am doing something wrong, but I discovered that by creating a blank block theme, applying only some styles via the Styles panel in the Site Editor, when I export or clone the theme using Create Block Theme, the theme.json doesn't seem to be updated with the new styles.

However, when I export the theme using the Site Editor Export option, the theme.json is updated as expected.

Steps to reproduce:

1. Create a block theme with only an empty `style.css` (except for the required header), `theme.json` with basic content (below), and a blank `templates/index.html` page.
2. Activate the theme, and only apply some styles like background image, text color, etc via Site Editor, and save.
3. Use the plugin page to either Export or Clone the theme

Original theme.json
```
{
  "version": 2,
  "settings": {
    "appearanceTools": true
  }
}
```
Exported/cloned theme.json

```
{
	"settings": {
		"appearanceTools": true
	},
	"version": 2
}
```

Expected theme.json (which I get via the Site Editor Export option).

```
{
	"$schema": "https://schemas.wp.org/wp/6.0/theme.json",
	"settings": {
		"appearanceTools": true
	},
	"styles": {
		"color": {
			"background": "var:preset|color|pale-cyan-blue",
			"text": "var:preset|color|vivid-cyan-blue"
		},
		"elements": {
			"link": {
				"color": {
					"text": "var:preset|color|vivid-purple"
				}
			}
		},
		"spacing": {
			"padding": {
				"bottom": "20px",
				"left": "20px",
				"right": "20px",
				"top": "20px"
			}
		}
	},
	"version": 2
}
```

**Comment by luminuu** [comment]
I just tested through this and none of the four options in my test saved the Global Styles I changed. Only the export directly from the site editor updated the theme.json file correctly.

**Comment by jonathanbossenger** [comment]
Update: after digging into this today, it looks like the problem is in the upstream code.

The `static::get_user_data()` method call at line [39 of the MY_Theme_JSON_Resolver class](https://github.com/WordPress/create-block-theme/blob/97ebac83af0a96670807cf9e3a38477bc255c630/admin/resolver_additions.php#L39) does return the global styles as saved in the database. 

Therefore I can only assume that the [merge](https://github.com/WordPress/wordpress-develop/blob/7573385d950ede32ba3feba48435d66d02a24766/src/wp-includes/class-wp-theme-json.php#L1570) method of the core WP_Theme_JSON class is where the problem exists.

I'll keep digging and report back.

**Comment by jonathanbossenger** [comment]
I think I've fixed this with #88 but I'd appreciate any additional testing to confirm.
